### ISSUE TITLE
Version 3.16.3: RedisTimeoutException: Command execution timeout for command: (PING)

### ISSUE THREAD

**Issue by olessio** [redisson/redisson#3867]
After switching to version 3.16.3, I see tons of next exceptions:
```
org.redisson.client.handler.PingConnectionHandler : Unable to send PING command over channel: [id: 0x76185fba, L:/127.0.0.1:59095 - R:localhost/127.0.0.1:6379]

 org.redisson.client.RedisTimeoutException: Command execution timeout for command: (PING), params: [], Redis client: [addr=redis://localhost:6379]
	at org.redisson.client.RedisConnection.lambda$async$1(RedisConnection.java:255)
	at io.netty.util.HashedWheelTimer$HashedWheelTimeout.expire(HashedWheelTimer.java:669)
	at io.netty.util.HashedWheelTimer$HashedWheelBucket.expireTimeouts(HashedWheelTimer.java:744)
	at io.netty.util.HashedWheelTimer$Worker.run(HashedWheelTimer.java:469)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

And I don't see the same in v3.16.2.

**Comment by mrniko** [comment]
fixed in https://github.com/redisson/redisson/commit/33e5b5872b098c13cf1bbf70f8152f7f3d94dc7c

**Comment by olessio** [comment]
Ok, so will wait for the next version.
Thanks for quick reply!

**Comment by olessio** [comment]
Testing in version 3.16.4 - the problem is still there.

**Comment by mrniko** [comment]
Should be fixed in https://github.com/redisson/redisson/commit/1c8bc8981cbbe1f978d0cd3eadb4cfccb13199cc

**Comment by olessio** [comment]
Thanks. I assume I need to wait for 3.16.5?

**Comment by mrniko** [comment]
Yes.

Which Redisson objects do you use?

**Comment by olessio** [comment]
Mostly I use streams/locks/collections.
I just see that I start my app and after some time of inactivity, it starts polluting the log with those exceptions above.
I've tested your fix - looks ok, so I'll wait for 3.16.5 (for now, I'll keep using 3.16.2)

**Comment by mrniko** [comment]
I recheck the fix and found it useless, since blocking command increments usage counter and this error doesn't happen.

**Comment by mrniko** [comment]
Can you share Redisson config?

**Comment by olessio** [comment]
See sample test below
I see that exception appearance depends on ConnectionMinimumIdleSize with the amount of streams we are reading from.
But I hope you can give more insights of why it happens and how to avoid it.
`
import java.time.Duration;
import java.util.Map;
import org.redisson.Redisson;
import org.redisson.api.RStream;
import org.redisson.api.RedissonClient;
import org.redisson.api.StreamMessageId;
import org.redisson.api.stream.StreamReadGroupArgs;
import org.redisson.client.codec.StringCodec;
import org.redisson.config.Config;

public class RedissonExceptionTest {
    public static void main(String[] args) {
        RedissonClient client = createClient();

        for (int i = 1; i <= 10; i++) {
            String streamName = "test.stream." + i;
            client.getKeys().delete(streamName);

            RStream<String, String> stream = client.getStream(streamName, StringCodec.INSTANCE);
            new Thread(new StreamReader(stream)).start();
        }
    }

    private static RedissonClient createClient() {
        Config config = new Config();
        config.useSingleServer()
            .setAddress("redis://localhost:6379")
            .setConnectionMinimumIdleSize(4)
            .setPingConnectionInterval(5000)
            .setPassword("changeme");
        return Redisson.create(config);
    }

    private static class StreamReader implements Runnable {
        private final RStream<String, String> stream;

        public StreamReader(RStream<String, String> stream) {
            this.stream = stream;
        }

        @Override
        public void run() {
            String groupName = stream + ".test";
            try {
                stream.createGroup(groupName, StreamMessageId.NEWEST);
            } catch (Exception exception) {
                //ignore
            }
            String consumer = groupName + ".consumer";
            while (true) {
                Map<StreamMessageId, Map<String, String>> data = stream.readGroup(
                    groupName,
                    consumer,
                    StreamReadGroupArgs.neverDelivered().timeout(Duration.ofMillis(5000)));
                if (data != null) {
                    System.out.println(stream.getName() + ": read " + data.size());
                    if (!data.isEmpty()) {
                        stream.ack(groupName, data.keySet().toArray(new StreamMessageId[0]));
                    }
                } else {
                    System.out.println("This is totally unexpected, but still happens sometimes after PING error");
                }
            }
        }
    }
}
`

**Comment by ZaNkss** [comment]
When i see this exception.CPU usage rises rapidly
Can you explain this phenomenon？
Can you explain why this exception occurs？
Thx

**Comment by mrniko** [comment]
@ZaNkss 

> When i see this exception.CPU usage rises rapidly
> Can you explain this phenomenon？
> Can you explain why this exception occurs？

It means that Redis doesn't respond in time on ping. Check slowlog result to get slow commands/scripts.

**Comment by mrniko** [comment]
Fixed.

@olessio 

Thanks for provided code

**Comment by nladuguie** [comment]
Hello, 
We also face this issue with following complete error and stacktrace : 
`Unable to send PING command over channel:
org.redisson.client.RedisTimeoutException: Command execution timeout for command: (PING), params: [], Redis client: [addr=redis://XXX:YYY] 	at org.redisson.client.RedisConnection.lambda$async$0(RedisConnection.java:245) 	at io.netty.util.HashedWheelTimer$HashedWheelTimeout.expire(HashedWheelTimer.java:669) 	at io.netty.util.HashedWheelTimer$HashedWheelBucket.expireTimeouts(HashedWheelTimer.java:744) 	at io.netty.util.HashedWheelTimer$Worker.run(HashedWheelTimer.java:469) 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) 	at java.base/java.lang.Thread.run(Thread.java:832)
--
`

We are mainly using Queues, locks and MapCache.
We are observing this error starts occurring and then our service is producing more and more error logs like the one provided.
Is this [commit](https://github.com/redisson/redisson/commit/69329c8dbdf8bf055bfd9951028d9250024d22e4) supposed to fix our use case ? 
Did someone bypassed this issue with specific configuration ?
As far as I can see, it's supposed to be shipped in 3.16.9, but do you have and ETA for that version to be released ?

Than you for your support.

**Comment by mrniko** [comment]
@nladuguie 

Next version will be released during this month.
